src/execution/operator/join/physical_range_join.cpp

```diff
@@ -227,8 +231,9 @@ idx_t PhysicalRangeJoin::LocalSortedTable::MergeNulls(const vector<JoinCondition
 		}
 	}
 
-	auto &primary = keys.data[0];
 	if (all_constant == keys.data.size()) {
+		// Make a copy of validity to avoid modifying original mask
```

Comment:
This doesn't make a copy - it sets the validity mask of the primary to the result of the constant NULL state. This works because if any of the constant keys are NULL then the row will be skipped, but the comment should reflect this. For this case I don't think you even need the mask - just the loop and return values?  It would help if there was a coverage test for this case.

Reply:
Thanks for you advice! I've added the tests and fixed the comment. The mask is still needed because the primary key is used for sorting later.